tests/suite/test_app_protect_grpc.py

```diff
@@ -0,0 +1,168 @@
+invalid_resp_text = "The request was rejected. Please consult with your administrator."
+
+class BackendSetup:
+    """
+    Encapsulate the example details.
+
+    Attributes:
+        ingress_host (str):
+    """
+
+    def __init__(self, ingress_host, ssl_port):
+        self.ingress_host = ingress_host
+        self.ssl_port = ssl_port
+
+
+@pytest.fixture(scope="function")
+def backend_setup(request, kube_apis, ingress_controller_endpoint, ingress_controller_prerequisites, test_namespace) -> BackendSetup:
+    """
+    Deploy a simple application and AppProtect manifests.
+
+    :param request: pytest fixture
+    :param kube_apis: client apis
+    :param ingress_controller_endpoint: public endpoint
+    :param test_namespace:
+    :return: BackendSetup
+    """
+    print("------------------------- Replace ConfigMap with HTTP2 -------------------------")
+    replace_configmap_from_yaml(kube_apis.v1,
+                            ingress_controller_prerequisites.config_map['metadata']['name'],
+                            ingress_controller_prerequisites.namespace,
+                            f"{TEST_DATA}/appprotect/grpc/nginx-config.yaml")
+
+    policy = request.param["policy"]
+    print("------------------------- Deploy backend application -------------------------")
+    create_example_app(kube_apis, "grpc", test_namespace)
+    wait_until_all_pods_are_ready(kube_apis.v1, test_namespace)
+
+    print("------------------------- Deploy Secret -----------------------------")
+    src_sec_yaml = f"{TEST_DATA}/appprotect/appprotect-secret.yaml"
+    create_items_from_yaml(kube_apis, src_sec_yaml, test_namespace)
+
+    print("------------------------- Deploy logconf -----------------------------")
+    src_log_yaml = f"{TEST_DATA}/appprotect/logconf.yaml"
+    log_name = create_ap_logconf_from_yaml(kube_apis.custom_objects, src_log_yaml, test_namespace)
+
+    print(f"------------------------- Deploy appolicy: {policy} ---------------------------")
+    src_pol_yaml = f"{TEST_DATA}/appprotect/grpc/{policy}.yaml"
+    pol_name = create_ap_policy_from_yaml(kube_apis.custom_objects, src_pol_yaml, test_namespace)
+
+    print("------------------------- Deploy Syslog -----------------------------")
+    src_syslog_yaml = f"{TEST_DATA}/appprotect/syslog.yaml"
+    create_items_from_yaml(kube_apis, src_syslog_yaml, test_namespace)
+    wait_before_test(40)
+    syslog_ep = (
+            kube_apis.v1.read_namespaced_endpoints("syslog-svc", test_namespace)
+            .subsets[0]
+            .addresses[0]
+            .ip
+        )
+    
+    print("------------------------- Deploy ingress -----------------------------")
+    ingress_host = {}
+    src_ing_yaml = f"{TEST_DATA}/appprotect/grpc/ingress.yaml"
+    create_ingress_with_ap_annotations(kube_apis, src_ing_yaml, test_namespace, policy, "True", "True", f"{syslog_ep}:514")
+    ingress_host = get_first_ingress_host_from_yaml(src_ing_yaml)
+    wait_before_test(40)
```

Comment:
we should be able to rely on an event here - once the IC reloads N+ successfully, it will emit an event for that Ingress resource. This will allow us avoid waiting for 40 seconds. However, since this `wait` approach is used in the existing test code, perhaps it makes sense to address it in a separate PR.

Reply:
The approach you suggested is the correct one and there's already a ticket in the backlog to fix this across the suite.